pkg/agent/proxy/proxier.go

```diff
@@ -513,21 +496,35 @@ func (p *proxier) installServices() {
 				}
 			}
 
-			updateInstalledEndpointsMap := func(endpoint k8sproxy.Endpoint) {
-				// If the Endpoint is newly installed, add a reference.
-				if _, ok := endpointsInstalled[endpoint.String()]; !ok {
+			if mcsLocalService != nil {
+				allReachableEndpoints = append(allReachableEndpoints, mcsLocalService.Endpoint)
+			}
+
```

Comment:
IMO, this function (`installServices`) is getting very long. Is there a way to extract some functionality into helper functions?

Reply:
Yes, it is quite long for this function,  I thought of extracting the code under `if needUpdateEndpoints` and `if needUpdateService` into functions, but I am not sure if this is an appropriate idea.